File: DungeonDebt/Assets/Scripts/Data/DifficultyPreset.cs
// Immutable run-contract preset. Assembled once in DataRepository from
// GameRules constants and applied once in RunManager.InitializeRun. The four
// combat multipliers are carried onto RunState this slice but not yet read by
// combat (M15.2).
public class DifficultyPreset
{
    public DifficultyPreset(
        DifficultyPresetId id,
        string displayName,
        int startingGold,
        int startingDebt,
        int startingMorale,
        int interestDivisor,
        int debtLimit,
        float heroHealthMult,
        float heroDamageMult,
        float enemyHealthMult,
        float enemyDamageMult)
    {
        Id = id;
        DisplayName = displayName;
        StartingGold = startingGold;
        StartingDebt = startingDebt;
        StartingMorale = startingMorale;
        InterestDivisor = interestDivisor;
        DebtLimit = debtLimit;
        HeroHealthMult = heroHealthMult;
        HeroDamageMult = heroDamageMult;
        EnemyHealthMult = enemyHealthMult;
        EnemyDamageMult = enemyDamageMult;
    }

    public DifficultyPresetId Id { get; }
    public string DisplayName { get; }
    public int StartingGold { get; }
    public int StartingDebt { get; }
    public int StartingMorale { get; }
    public int InterestDivisor { get; }
    public int DebtLimit { get; }
    public float HeroHealthMult { get; }
    public float HeroDamageMult { get; }
    public float EnemyHealthMult { get; }
    public float EnemyDamageMult { get; }
}

File: web/src/data/DifficultyPreset.js
// Ported from DungeonDebt/Assets/Scripts/Data/DifficultyPreset.cs
export class DifficultyPreset {
  constructor(id, displayName, startingGold, startingDebt, startingMorale, interestDivisor, debtLimit, heroHealthMult, heroDamageMult, enemyHealthMult, enemyDamageMult) {
    this.id = id;
    this.displayName = displayName;
    this.startingGold = startingGold;
    this.startingDebt = startingDebt;
    this.startingMorale = startingMorale;
    this.interestDivisor = interestDivisor;
    this.debtLimit = debtLimit;
    this.heroHealthMult = heroHealthMult;
    this.heroDamageMult = heroDamageMult;
    this.enemyHealthMult = enemyHealthMult;
    this.enemyDamageMult = enemyDamageMult;
    Object.freeze(this);
  }
}
